File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalPPMVisit/QuoteContractOriginalPPMVisit.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteContractOriginalPPMVisits

        Public Class QuoteContractOriginalPPMVisit

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "QuoteContractPPMVisit Properties"


            Private _QuoteContractPPMVisitID As Integer = 0
            Public ReadOnly Property QuoteContractPPMVisitID() As Integer
                Get
                    Return _QuoteContractPPMVisitID
                End Get
            End Property
            Public WriteOnly Property SetQuoteContractPPMVisitID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteContractPPMVisitID", Value)
                End Set
            End Property


            Private _QuoteContractSiteID As Integer = 0
            Public ReadOnly Property QuoteContractSiteID() As Integer
                Get
                    Return _QuoteContractSiteID
                End Get
            End Property
            Public WriteOnly Property SetQuoteContractSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_QuoteContractSiteID", Value)
                End Set
            End Property


            Private _EstimatedVisitDate As DateTime = Nothing
            Public Property EstimatedVisitDate() As DateTime
                Get
                    Return _EstimatedVisitDate
                End Get
                Set(ByVal Value As DateTime)
                    _EstimatedVisitDate = Value
                End Set
            End Property



#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalPPMVisit/QuoteContractOriginalPPMVisit.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace QuoteContractOriginalPPMVisits
    {
        public class QuoteContractOriginalPPMVisit
        {
            private DataTypeValidator _dataTypeValidator;

            public QuoteContractOriginalPPMVisit()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _QuoteContractPPMVisitID = 0;

            public int QuoteContractPPMVisitID
            {
                get
                {
                    return _QuoteContractPPMVisitID;
                }
            }

            public object SetQuoteContractPPMVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteContractPPMVisitID", value);
                }
            }

            private int _QuoteContractSiteID = 0;

            public int QuoteContractSiteID
            {
                get
                {
                    return _QuoteContractSiteID;
                }
            }

            public object SetQuoteContractSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_QuoteContractSiteID", value);
                }
            }

            private DateTime _EstimatedVisitDate = default;

            public DateTime EstimatedVisitDate
            {
                get
                {
                    return _EstimatedVisitDate;
                }

                set
                {
                    _EstimatedVisitDate = value;
                }
            }



            
        }
    }
}